File: 0401_0500/0413/0413.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n, m = (int(x) for x in f.readline().split())
    p = [f.readline().strip() for i in range(n)]
    s = sum(1 if p[i][j] == '#' and (i == 0 or p[i - 1][j] == '.') and (j == 0 or p[i][j - 1] == '.') else 0 for j in range(m) for i in range(n))
    q.write(str(s))

File: 0401_0500/0413/0413.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n, m, k = 0;
	char **s;

	fscanf(f, "%d%d", &n, &m);
	s = (char**)malloc(sizeof(char*) * n);
	for (int i = 0; i < n; ++i)
	{
		s[i] = (char*)malloc(m);
		fscanf(f, "%s\n", s[i]);
	}
	for (int i = 0; i < n; ++i)
		for (int j = 0; j < m; ++j)
			if (s[i][j] == '#' && (i == 0 || s[i - 1][j] == '.') && (j == 0 || s[i][j - 1] == '.'))
				++k;
	fprintf(q, "%d", k);
	return 0;
}
